benchmarks/eval_vlm_perplexity.py

```
@@ -0,0 +1,187 @@
+    p = _P(spec)
+    if p.is_dir():
+        shards = sorted(p.glob("*.safetensors"))
+        return {"spec": str(spec), "kind": "local",
+                "safetensors_bytes": sum(f.stat().st_size for f in shards),
+                "n_shards": len(shards)}
+    try:
+        from huggingface_hub import HfApi
+
+        return {"spec": str(spec), "kind": "hub",
+                "revision": HfApi().model_info(str(spec)).sha}
+    except Exception as exc:  # offline, private, or not a repo id
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)